source/deprecation-policy.rst

```diff
@@ -0,0 +1,145 @@
+---
+layout: default
+title: Deprecation Policy
+menuSlug: deprecation-policy
+permalink: /deprecation-policy/index.html
+---
+
+Deprecation Policy
+==================
+
+The Doctrine team will adhere to this deprecation policy whenever it is reasonably possible.
+
+Using ``@deprecated`` and ``PHPStan Deprecation Rules``
+-------------------------------------------------------
+
+The primary and preferred method for deprecating functionality within Doctrine is by using
+the ``@deprecated`` annotation. To have usages of functionality in your dependencies analyzed
+and reported, you must use a tool like `PHPStan Deprecation Rules`_ or `Pslam Static Analysis`_.
+
+Here is an example of how this would be used in Doctrine:
+
+.. code-block:: php
+
+    namespace Doctrine\ORM;
+
+    class EntityManager
+    {
+        // ...
+
+        /**
+         * @deprecated
+         */
+        public function getSomething() : int
+        {
+            // ...
+        }
+    }
+
+Then, if your code directly depends on ``doctrine/orm`` which provides the
+``Doctrine\ORM\EntityManager`` class, you will get a warning about it when you run
+`PHPStan Deprecation Rules`_ or `Pslam Static Analysis`_. Here is an example:
+
+.. code-block:: php
+
+    namespace App;
+
+    use App\Entities\User;
+    use Doctrine\ORM\EntityManager;
+
+    class UserRegistration
+    {
+        /** @var EntityManager */
+        private $entityManager;
+
+        // ...
+
+        public function create(array $userData) : User
+        {
+            $something = $this->entityManager->getSomething();
+
+            // ...
+        }
+    }
+
+Using ``trigger_error()`` and ``phpunit-bridge``
+------------------------------------------------
+
+In some cases, you may need to conditionally deprecate functionality. In these cases, it is required
+to use the ``trigger_error()`` function and a tool like `PHPUnit Bridge`_ which implements a custom
+error handler for you to use to collect and report these warnings. This method is not recommended
```

Comment:
> This method is not recommended and Doctrine will try to avoid using it if possible.

What do you think about making this bold? Just to emphasise a bit more :+1: